Sources/Sentry/SentryRetryAfterHeaderParser.m

```diff
@@ -0,0 +1,45 @@
+#import <Foundation/Foundation.h>
+#import "SentryRetryAfterHeaderParser.h"
+#import "SentryCurrentDate.h"
+#import "SentryHttpDateParser.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryRetryAfterHeaderParser ()
+
+@property(nonatomic, strong) SentryHttpDateParser *httpDateParser;
+
+@end
+
+@implementation SentryRetryAfterHeaderParser
+
+- (instancetype)initWithHttpDateParser:(SentryHttpDateParser *)httpDateParser {
+    if (self = [super init]) {
+        self.httpDateParser = httpDateParser;
+    }
+    return self;
+}
+
+- (NSDate *_Nullable)parse:(NSString *_Nullable)retryAfterHeader {
+    if (nil == retryAfterHeader || 0 == [retryAfterHeader length]) {
+        return nil;
+    }
+
+    NSDate *now = [SentryCurrentDate date];
+
+    // try to parse as double/seconds
+    double retryAfterSeconds = [retryAfterHeader doubleValue];
+    NSLog(@"parseRetryAfterHeader string '%@' to double: %f", retryAfterHeader, retryAfterSeconds);
```

Comment:
That is a leftover from the old code. I just moved it there. Now that I think about it I'm going to remove the `NLSLog`, because we don't need it IMO. Next time I'm going to use a logger I'm going to do it like you suggested. Good point 👍